### PR TITLE
AO3-4370 Performance : follow the advice of bullet

### PR DESCRIPTION
**Pull request by zz9pzza**
Bullet gives advice to reduce the number of calls to the database server. Given at present we run at about 45 database calls per page. Anything we can do to reduce it is a good thing. The queries will be harder so the average time to run each query will go up a bit.

This will also mean others can see the warnings when they are developing.

### CODE REVIEW

**Comment by shalott** [comment]
The failures in the collections feature seem to be real failures having to do with loading the challenge object via polymorphic association?

**Comment by sarken** [comment]
I haven't reviewed the changes @shalott requested, but I wanted to note that this is now unmergable.

**Comment by shalott** [comment]
The issue I spotted is fixed however so once the merge fixes are done I am happy for this to be marked ready-to-merge, this looks v useful

File: config/initializers/active_record_log_subscriber.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module LogQuerySource
```
Comment (elzj) [inline config/initializers/active_record_log_subscriber.rb]: @zz9pzza Can you add some comments to this? What are we changing and why are we changing it?

Comment (zz9pzza) [inline config/initializers/active_record_log_subscriber.rb]: This comes from http://www.jkfill.com/2015/02/14/log-which-line-caused-a-query/ 

And makes it significantly easier to find out where the issue is ( but not where is the right place to fix it )

**Comment by zz9pzza** [comment]
We plan to put this in to multiple pull requests

**Comment by zz9pzza** [comment]
Closing as I have broken this down in to a number of pull request

**Pull request closed without merging**